packages/flutter_tools/test/test_compiler_test.dart

```diff
@@ -2,28 +2,38 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
-import 'package:file/memory.dart';
 import 'package:flutter_tools/src/base/file_system.dart';
 import 'package:flutter_tools/src/compile.dart';
 import 'package:flutter_tools/src/project.dart';
 import 'package:flutter_tools/src/test/test_compiler.dart';
 import 'package:mockito/mockito.dart';
 
 import 'src/common.dart';
-import 'src/context.dart';
+import 'src/testbed.dart';
 
 void main() {
-  group('TestCompiler', () {
-    testUsingContext('compiles test file with no errors', () async {
-      fs.file('pubspec.yaml').createSync();
-      fs.file('.packages').createSync();
-      fs.file('test/foo.dart').createSync(recursive: true);
-      final MockResidentCompiler residentCompiler = MockResidentCompiler();
-      final TestCompiler testCompiler = FakeTestCompiler(
-        false,
-        await FlutterProject.current(),
-        residentCompiler,
+  group(TestCompiler, () {
+    Testbed testbed;
+    FakeTestCompiler testCompiler;
+    MockResidentCompiler residentCompiler;
+
+    setUp(() {
+      testbed = Testbed(
```

Comment:
In theory this could be replaced with a regular setUp function by replacing `fs` with an instance of MemoryFileSystem, and passing that to testUsingContext. However, you need to be careful that none of that code transitively calls any injected values, or else the test state will leak.

The testbed simplifies this by providing a setUp function that is run in the same zone that the test will be run in.